code/datums/chatmessage.dm

```diff
@@ -227,23 +236,31 @@ GLOBAL_LIST_INIT(job_colors_pastel, list(
 	if (current_z_idx >= CHAT_LAYER_MAX_Z)
 		current_z_idx = 0
 
+	var/bound_height = world.icon_size
+	var/bound_width = world.icon_size
+	if(ismovableatom(message_loc))
+		var/atom/movable/AM = message_loc
+		bound_height = AM.bound_height
+		bound_width = AM.bound_width
 	// Build message image
 	message = image(loc = message_loc, layer = CHAT_LAYER + CHAT_LAYER_Z_STEP * current_z_idx++)
 	message.plane = RUNECHAT_PLANE
 	message.appearance_flags = APPEARANCE_UI_IGNORE_ALPHA | KEEP_APART
 	message.alpha = 0
-	message.pixel_y = owner.bound_height - MESSAGE_FADE_PIXEL_Y
+	message.pixel_y = bound_height - MESSAGE_FADE_PIXEL_Y
 	message.maptext_width = CHAT_MESSAGE_WIDTH
 	message.maptext_height = mheight
-	message.maptext_x = (CHAT_MESSAGE_WIDTH - owner.bound_width) * -0.5
+	message.maptext_x = (CHAT_MESSAGE_WIDTH - bound_width) * -0.5
 	if(extra_classes.Find("italics"))
 		message.color = "#CCCCCC"
 	message.maptext = MAPTEXT(complete_text)
 
-	// View the message
-	LAZYADDASSOCLIST(owned_by.seen_messages, message_loc, src)
-	owned_by.images |= message
-	animate(message, alpha = 255, pixel_y = owner.bound_height, time = CHAT_MESSAGE_SPAWN_TIME)
+	// Show the message to clients
+	for(var/client/C in hearers)
+		C?.images |= message
```

Comment:
Just lazy add here

Reply:
Its not a lazy list

Reply:
Also lazy wouldn't handle C being null, it would only handle images being null.